app/models/student.py

```diff
@@ -72,12 +84,23 @@ def put(sid):
         return student
         
     
-    
+    @staticmethod
     @marshal_with(studentFields)
-    def delete(sid):
+    def delete_student_by_id(sid):
         student=SMS.query.filter_by(sid=sid).first()
         db.session.delete(student)
         db.session.commit()
         students=SMS.query.all()
         
-        return students
\ No newline at end of file
+        return students
+    
+    @marshal_with(studentFields)
+    def search_student_by_name(name):
```

Comment:
Search functionality should be a part of get_student() method. @siraj-shaikh-BS

Reply:
Okay